src/components/containers/DashboardContainer.js

```diff
@@ -4,18 +4,45 @@ import { connect } from 'react-refetch';
 import Spinner from '../views/Spinner';
 import ErrorComponent from '../views/Error';
 import Dashboard from '../views/Dashboard';
+import URLManager from '../../lib/urlManager';
 
 
 class DashboardContainer extends React.Component {
     constructor(props) {
         super(props);
-        this.state = { activeCategory: null };
+
+        // Only apply a region that is set as a query parameter. Session storage
+        // also records preferred regions, but the presence of a region in
+        // session storage should not be seen as a sign that the region needs to
+        // be applied to this dashboard.
+        const activeRegion = URLManager.getQueryParameter('region');
+        this.state = { activeRegion };
     }
 
-    _onCategoryChange = e => {
-        this.setState({
-            activeCategory: e.target.value,
-        });
+    _onRegionChange = e => {
+        const newRegion = e.target.value;
+
+        // Record the query paramater in the URL so that users can bookmark the
+        // dashboard at this region or send others a link to the dashboard at
+        // this region
+        URLManager.setQueryParameter('region', newRegion);
+
+        // Record the region in session storage as well.
+        //
+        // Pages that do not support regions do not get a ?region query
+        // paramater so as to avoid confusion. But we still need a way to
+        // remember the user's preferred region so that it can be used when the
+        // user returns to a page that does support regions.
+        sessionStorage.setItem('region', newRegion);
+
+        // Dispatch a custom event which other components can listen for if they
+        // need to respond to region changes
+        const root = document.getElementById('root');
+        const regionChangedEvent = document.createEvent('Event');
+        regionChangedEvent.initEvent('regionChanged', true, true);
+        root.dispatchEvent(regionChangedEvent);
```

Comment:
I like the pub/sub approach though as I recall custom events can be a cross-browser pain. I imagine this should work most places but have lost faith in IE especially.

Reply:
MDN says that this approach to custom events has wide browser support, and it worked in IE11 on BrowserStack, but this is a good point and I'll do more testing.